cloudinit/config/cc_raspberry_pi.py

```diff
@@ -0,0 +1,211 @@
+                "nonint",
+                RASPI_CONFIG_SERIAL_CONS_FN,
+                str(0 if enable_console else 1),
+            ]
+        )
+
+        try:
+            subp.subp(
+                [
+                    "/usr/bin/raspi-config",
+                    "nonint",
+                    RASPI_CONFIG_SERIAL_HW_FN,
+                    str(0 if enable_hw else 1),
+                ]
+            )
+        except subp.ProcessExecutionError as e:
+            LOG.error("Failed to configure serial hardware: %s", e)
+
+        require_reboot(instCfg)
+    except subp.ProcessExecutionError as e:
+        LOG.error("Failed to configure serial console: %s", e)
+
+
+def configure_interface(iface: str, enable: bool) -> None:
+    assert (
+        iface in SUPPORTED_INTERFACES.keys() and iface != "serial"
+    ), f"Unsupported interface: {iface}"
+
+    try:
+        subp.subp(
+            [
+                "/usr/bin/raspi-config",
+                "nonint",
+                SUPPORTED_INTERFACES[iface],
+                str(0 if enable else 1),
+            ]
+        )
+    except subp.ProcessExecutionError as e:
+        LOG.error("Failed to configure %s: %s", iface, e)
+
+
+def handle(name: str, cfg: Config, cloud: Cloud, args: list) -> None:
+    if RPI_BASE_KEY not in cfg:
+        return
+    elif not isinstance(cfg[RPI_BASE_KEY], dict):
+        LOG.warning(
+            "Invalid value for %s: %s",
+            RPI_BASE_KEY,
+            cfg[RPI_BASE_KEY],
+        )
+        return
+    elif not cfg[RPI_BASE_KEY]:
+        LOG.debug("Empty value for %s. Skipping...", RPI_BASE_KEY)
+        return
+
+    for key in cfg[RPI_BASE_KEY]:
+        if key == ENABLE_RPI_CONNECT_KEY:
+            enable = cfg[RPI_BASE_KEY][key]
+
+            if isinstance(enable, bool):
+                configure_rpi_connect(enable)
+            else:
+                LOG.warning(
+                    "Invalid value for %s: %s", ENABLE_RPI_CONNECT_KEY, enable
+                )
+            continue
+        elif key == RPI_INTERFACES_KEY:
+            if not isinstance(cfg[RPI_BASE_KEY][key], dict):
+                LOG.warning(
+                    "Invalid value for %s: %s",
+                    RPI_BASE_KEY,
+                    cfg[RPI_BASE_KEY][key],
+                )
+                return
+            elif not cfg[RPI_BASE_KEY][key]:
+                LOG.debug("Empty value for %s. Skipping...", key)
+                return
+
+            # check for supported ARM interfaces
+            for subkey in cfg[RPI_BASE_KEY][key]:
+                if subkey not in SUPPORTED_INTERFACES.keys():
+                    LOG.warning(
+                        "Invalid key for %s: %s", RPI_INTERFACES_KEY, subkey
+                    )
+                    continue
+
+                enable = cfg[RPI_BASE_KEY][key][subkey]
+
+                if subkey == "serial":
+                    if not isinstance(enable, dict) and not isinstance(
```

Comment:
This can be simplified to `if not isinstance(enable, (dict, bool))`

Reply:
Thanks for the hint